core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/EraSubmissionPreparation.java

```diff
@@ -0,0 +1,496 @@
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.TransformerFactoryConfigurationError;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import com.eaglegenomics.simlims.core.User;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Experiment;
+import uk.ac.bbsrc.tgac.miso.core.data.Experiment.RunPartition;
+import uk.ac.bbsrc.tgac.miso.core.data.Index;
+import uk.ac.bbsrc.tgac.miso.core.data.Library;
+import uk.ac.bbsrc.tgac.miso.core.data.Pair;
+import uk.ac.bbsrc.tgac.miso.core.data.Partition;
+import uk.ac.bbsrc.tgac.miso.core.data.Pool;
+import uk.ac.bbsrc.tgac.miso.core.data.Run;
+import uk.ac.bbsrc.tgac.miso.core.data.Sample;
+import uk.ac.bbsrc.tgac.miso.core.data.Study;
+import uk.ac.bbsrc.tgac.miso.core.data.Submission;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.view.PoolableElementView;
+import uk.ac.bbsrc.tgac.miso.core.data.type.HealthType;
+import uk.ac.bbsrc.tgac.miso.core.data.type.SubmissionActionType;
+
+public class EraSubmissionPreparation {
+  private abstract class ChildSubmissionFile {
+    public String fileName() {
+      return String.format("%s_%s_%s.xml", submission.getAlias(), name(), DF_TIMESTAMP.format(submission.getSubmissionDate()));
+    }
+
+    public abstract void generateDocument(ZipOutputStream output)
+        throws ParserConfigurationException, IOException, TransformerConfigurationException, TransformerException,
+        TransformerFactoryConfigurationError;
+
+    public abstract boolean isEmpty();
+
+    public abstract String name();
+  }
+
+  private final class DilutionXmlSubfile extends XmlSubmissionFromSet<Pair<PoolableElementView, Partition>> {
+
+    @Override
+    protected Stream<Pair<PoolableElementView, Partition>> items() {
+      return submission.getExperiments().stream().flatMap(experiment -> experiment.getRunPartitions().stream()).flatMap(
+          rp -> rp.getPartition().getPool().getPoolableElementViews().stream().map(poolable -> new Pair<>(poolable, rp.getPartition())));
+    }
+
+    @Override
+    public String name() {
+      return "run";
+    }
+
+    @Override
+    protected void populate(Element xml, Pair<PoolableElementView, Partition> entry) {
+      Run r = entry.getValue().getSequencerPartitionContainer().getLastRun();
+
+      xml.setAttribute("alias",
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "alias" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)